src/network/networkd-wifi.c

```diff
@@ -90,6 +90,18 @@ int manager_genl_process_nl80211_config(sd_netlink *genl, sd_netlink_message *me
 
         r = link_get_by_index(manager, ifindex, &link);
         if (r < 0) {
+                /* The NL80211_CMD_NEW_INTERFACE message might arrive before RTM_NEWLINK, in which case a
+                 * link will not have been created yet. Store the interface index such that the wireless
+                 * properties of the link (such as wireless interface type) are queried again after the link
+                 * is created.
+                 */
+                if (cmd == NL80211_CMD_NEW_INTERFACE) {
+                        r = set_ensure_put(&manager->new_wlan_ifindices, NULL, INT_TO_PTR(ifindex));
```

Comment:
This updates `r` to be used by the debugging logs below. Please introduce another variable here:
```c
int q = set_ensure_put(...);
if (q < 0)
        log_warning_errno(q, "Failed to ...");
```